src/daemon/plugin.c

```diff
@@ -1707,13 +1704,9 @@ EXPORT int plugin_init_all(void) {
    * callback. */
   le = llist_head(list_init);
   while (le != NULL) {
-    callback_func_t *cf;
-    plugin_init_cb callback;
-    plugin_ctx_t old_ctx;
-
-    cf = le->value;
-    old_ctx = plugin_set_ctx(cf->cf_ctx);
-    callback = cf->cf_callback;
+    callback_func_t *cf = le->value;
+    plugin_ctx_t old_ctx = plugin_set_ctx(cf->cf_ctx);
+    plugin_init_cb callback = (void *)cf->cf_callback;
```

Comment:
Same here as below regarding the type: shouldn't the cast be more specific instead of `(void *)`?

Reply:
The C standard says that all pointer types can be case to `void*`, and `void*` can be cast to any pointer type. Direct pointer conversion sometimes leads to compilers complaining, so I felt it safest to use the standard-approved method.

In retrospect, it didn't remove the warnings emitted by the Sun compiler, and I think it didn't have any effect on the result. If you prefer, I can revert those changes …?

Reply:
As this is for enhances compatibility this sounds reasonable.